pkg/utils/utils.go

```diff
@@ -64,3 +76,13 @@ func ParseSubjectReference(subRef string) (common.Reference, error) {
 func TrimSpaceAndToLower(input string) string {
 	return strings.ToLower(strings.TrimSpace(input))
 }
+
+// ParseRequestKey parses key string to a structured RequestKey object.
+func ParseRequestKey(key string) RequestKey {
+	re := regexp.MustCompile(subjectPattern)
+	match := re.FindStringSubmatch(key)
```

Comment:
I think we need a nil check here: the golang docstring shows that the function returns a nil if no match https://pkg.go.dev/regexp#Regexp.FindStringSubmatch